README.md

```diff
@@ -429,17 +429,37 @@ misbehavingStream.log();
 
 ### Future
 
-#### `Future#listen(o: Consumer<A>): void`
+#### `Future.of(a: A): Future<A>`
 
-Adds a consumer as listener to a future. If the future has already
-occurred the consumer is immediately pushed to.
+Converts any value into a future that happens immediately.
```

Comment:
I'll have to be a bit nitpicky about the wording here.

Semantically `Future.of(a)` is equivalent to `(-Infinity, a)`. This means that `Future.of` returns a future that has always occurred. The wording "happens immediately" to me sounds as if `Future.of` returns a future that occurs at the time where `Future.of` is called. This however, would not be a pure function.

I hope that makes sense. I think the wording should be changed to something like

> Returns a future that has already occurred with the given value.

Or maybe

> Returns a future that has at any point in time always occurred with the given value.

 However, these may be to "technical" or unintuitive. So let me know if you have a better idea :smile:

Reply:
Ah, makes sense. Thanks! I much prefer to over-explain than under-explain a concept, so I'll try to incorporate a lot of what you said above into the docs...